Makefile

```diff
@@ -0,0 +1,37 @@
+COLOUR_GREEN=\033[0;32m
+COLOUR_RED=\033[0;31m
+COLOUR_BLUE=\033[0;34m
+COLOUR_END=\033[0m
+
+.DEFAULT_GOAL:=help
+
+SHELL := /bin/bash
+GENERAL_LIB_LOCATION := $(shell pip show rados | grep -oP "(?<=Location: ).*")
+LIBRADOS_LOCATION := $(shell find "${GENERAL_LIB_LOCATION}" -name 'rados*.so' -print)
+LIBRADOS_EGGINFO_LOCATION := $(shell find "${GENERAL_LIB_LOCATION}" -name 'rados*.egg-info' -print)
+LIBRADOS_NAME := $(shell basename "${LIBRADOS_LOCATION}")
+LIBRADOS_EGGINFO_NAME := $(shell basename "${LIBRADOS_EGGINFO_LOCATION}")
+
+.PHONY: help
+help: ## Display this help message
+	@echo -e '${COLOUR_RED}Usage: make <command>${COLOUR_END}'
+	@cat $(MAKEFILE_LIST) | grep '^[a-zA-Z]'  | \
+	    awk -F ':.*?## ' 'NF==2 {printf "  %-26s%s\n\n", $$1, "${COLOUR_GREEN}"$$2"${COLOUR_END}"}'
+
+.PHONY: setup
+setup: ## Setting up the pre-commit environment and then the venv environment
```

Comment:
You can write setup: setup-pre-commit setup-venv ## Setup the local development environment
(I changed the comment, as setup will probably do more later, and simple present instead of present continuous)

Reply:
If you do so, you don't need any implementation for the setup-target itself, because setup-pre-commit and setup-venv are dependencies for setup

Reply:
ok then did that